src/main/scala/is/hail/io/bgen/BgenRecord.scala

```diff
@@ -116,13 +117,46 @@ class Bgen12GenotypeIterator(a: Array[Byte],
 
   private val pa = new BGen12ProbabilityArray(a, nSamples, nGenotypes, nBitsPerProb)
 
+  def sampleMissing(s: Int): Boolean = (a(8 + s) & 0x80) != 0
```

Comment:
consider changing to `isMissingSample` to indicate Boolean value, pairs will with `missingSamples`